File: SistemaDecomisos/src/java/model/Policia.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.io.Serializable;

/**
 *
 * @author Marco
 */
public class Policia extends Persona implements Serializable, Jsonable{
    private int idPolicia;

    public Policia() {
    }

    public Policia(int idPolicia, String identificacion, String nombre, String apellido1, String apellido2) {
        super(identificacion,nombre,apellido1,apellido2);
        this.idPolicia = idPolicia;
    }

    public int getIdPolicia() {
        return idPolicia;
    }

    public void setIdPolicia(int idPolicia) {
        this.idPolicia = idPolicia;
    }
    
}

File: web/js/objects/Policia.js
function Policia(identificacion, puesto, nombre, apellido1, apellido2, idPolicia) {
    this.Policia(identificacion, puesto, nombre, apellido1, apellido2, idPolicia);
}

Policia.prototype = {
    identificacion: "",
    puesto: "",
    nombre: "",
    apellido1: "",
    apellido2: "",
    idPolicia: 0,
    Policia: function (identificacion, puesto, nombre, apellido1, apellido2, idPolicia) {
        this.identificacion = identificacion;
        this.puesto = puesto;
        this.nombre = nombre;
        this.apellido1 = apellido1;
        this.apellido2 = apellido2;
        this.idPolicia = idPolicia;
    }
};

Policia.from = function (plain) {
    policia = new Policia(
            plain.identificacion,
            plain.puesto,
            plain.nombre,
            plain.apellido1,
            plain.apellido2,
            plain.idPolicia);
    return policia;
};

Policia.to = function (policia) {
    return {
        _class: 'Policia',
        identificacion: policia.idPolicia,
        puesto: policia.puesto,
        nombre: policia.nombre,
        apellido1: policia.apellido1,
        apellido2: policia.apellido2,
        idPolicia: policia.idPolicia
    };
};
 